uxarray/grid/grid.py

```diff
@@ -7,87 +7,112 @@
-        Input ``xarray.Dataset`` or vertex coordinates that form faces.
-
-    Other Parameters
-    ----------------
-    gridspec: bool, optional
-        Specifies gridspec
-    islatlon : bool, optional
-        Specify if the grid is lat/lon based
-    isconcave: bool, optional
-        Specify if this grid has concave elements (internal checks for this are possible)
-    use_dual: bool, optional
-        Specify whether to use the primal (use_dual=False) or dual (use_dual=True) mesh if the file type is mpas
-
-    Raises
-    ------
-        RuntimeError
-            If specified file not found or recognized
+    grid_ds : xr.Dataset
+        ``xarray.Dataset`` encoded in the UGRID conventions
+
+    source_grid_spec : str, default="UGRID"
+        Original unstructured grid format (i.e. UGRID, MPAS, etc.)
+
+    source_dims_dict : dict, default={}
+        Mapping of dimensions from the source dataset to their UGRID equivalent (i.e. {nCell : nMesh2_face})
 
     Examples
     ----------
 
     >>> import uxarray as ux
+    >>> grid_path = "/path/to/grid.nc"
+    >>> data_path = "/path/to/data.nc"
 
     1. Open a grid file with `uxarray.open_grid()`:
 
-    >>> uxgrid = ux.open_grid("filename.g")
+    >>> uxgrid = ux.open_grid(grid_path)
 
     2. Open an unstructured grid dataset file with
-    `uxarray.open_dataset()`, then access `Grid` info:
+    `uxarray.open_dataset()`, then access the ``Grid``.:
 
-    >>> uxds = ux.open_dataset("filename.g")
+    >>> uxds = ux.open_dataset(grid_path, data_path)
+    >>> uxds.uxgrid
     """
 
-    def __init__(self, input_obj, **kwargs):
-        # initialize internal variable names
-        self.__init_grid_var_names__()
-
-        # initialize face_area variable
-        self._face_areas = None
+    def __init__(self,
+                 grid_ds: xr.Dataset,
+                 source_grid_spec: Optional[str] = None,
+                 source_dims_dict: Optional[dict] = {}):
+
+        # check if inputted dataset is a minimum representable 2D UGRID unstructured grid
+        _is_minimum_ugrid = _validate_minimum_ugrid(grid_ds)
+
+        if not _is_minimum_ugrid:
+            raise ValueError(
+                "Directly constructing a Grid requires the input dataset to follow the UGRID conventions as "
+                "described in our documentation."
+            )  # TODO: elaborate once we have a formal definition
+
+        # grid spec not provided, check if grid_ds is a minimum representable UGRID dataset
+        if source_grid_spec is None:
+            warn(
+                "Attempting to construct a Grid without passing in source_grid_spec. Use of Grid constructor"
+                "is only advised if grid_ds is following the internal unstructured grid definition, including"
+                "variable and dimension names. Using ux.open_grid() or ux.from_dataset() is suggested.",
+                Warning)
+            if _is_minimum_ugrid:
```

Comment:
At this point, it's guaranteed that `_is_minimum_ugrid `, because we passed the exception above at Line 87,

Reply:
Should be fixed now!

Reply:
What I meant was to replace Lines 83-102 with this (note I also rephrased the text of the ValueError):

       ```
        # check if inputted dataset is a minimum representable 2D UGRID unstructured grid
        if not _validate_minimum_ugrid(grid_ds):
            raise ValueError(
                "Use of Grid constructor requires grid_ds to follow the internal unstructured grid definition, 
                  including variable and dimension names. This grid_ds is not satisfying those requirements. If you are 
                  not sure about how to do that, using ux.open_grid() or ux.from_dataset() is suggested."
            )  # TODO: elaborate once we have a formal definition

        # grid spec not provided, check if grid_ds is a minimum representable UGRID dataset
        if source_grid_spec is None:
            warn(
                "Attempting to construct a Grid without passing in source_grid_spec. Use of Grid constructor"
                "is only advised if grid_ds is following the internal unstructured grid definition, including"
                "variable and dimension names. Using ux.open_grid() or ux.from_dataset() is suggested.",
                Warning)
                
           source_grid_spec = "UGRID"
           source_dims_dict = None
```

Reply:
Additionally, You had  `source_grid_spec = "UGRID"` but I don't think we should do it since we can't know the type of the source grid unless the user states.

Reply:
> Additionally, You had source_grid_spec = "UGRID" but I don't think we should do it since we can't know the type of the source grid unless the user states.

If it passes the minimum UGRID, should we not set it as ugrid?

Reply:
Yes, I think we should not set it as UGRID. Think about the `Grid.from_dataset()` case: In Line 140:

`source_grid_spec = _parse_grid_type(dataset)`

`_parse_grid_type()` is recognizing the grid-spec from the input `dataset`, not our constructed `grid_ds ` i.e. we are setting `source_grid_spec` to the original mesh type, not UGRID. 

Similaryl, in this constructor, if the user prefers to specify what was the original grid spec they created the UGRID from, they will set `source_grid_spec`; i.e. if we are not given that info, we should keep it as `None`.

Hope this helps

Reply:
It does help! Makes a lot of sense! Thank you!